File: Algorithms/389 - Find the Difference.py
class Solution(object):
    def findTheDifference(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: str
        """
        
        s = sorted(s)
        t = sorted(t)
        
        i = 0
        while i < len(s):
            if t[i] != s[i]:
                return t[i]
            i += 1
        return t[len(t) - 1]

File: Algorithms/cpp/389 - Find the Difference.cpp
class Solution {
public:
    char findTheDifference(string s, string t) {
        // var
        vector<int> count(26, 0);
        
        // init
        for (int i = 0; i < s.length(); i++)
            count[s[i] - 'a']++;
        
        for (int i = 0; i < t.length(); i++) {
            if (count[t[i] - 'a'] == 0)
                return t[i];
            count[t[i] - 'a']--;
        }
    }
};